### PR TITLE
QH - DSPDC-804 - Stage a release_history object in GCS

### PR DESCRIPTION
**Pull request by quazi-broad**
Added a new template - create-json-object to save release date and archive path (provided as params) to a json file and then rsnyc the file to the provided upload path.
Need to confirm that this is working as expected.

### CODE REVIEW

File: orchestration/templates/create-json-object.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+          - name: upload-path
+      {{- include "argo.retry" . | indent 6 }}
+      script:
+        image: google/cloud-sdk:slim
+        command: [bash]
+        source: |
+        json='{"release_date":"'{{ "{{inputs.parameters.release-date}}" }}'","archive_path":"'{{ "{{inputs.parameters.archive-path}}" }}'"}'
+        echo "$json" > release_history.json
+        gsutil -m rsync -r /state/release_history.json {{ "{{inputs.parameters.upload-path}}" }}
+        volumeMounts:
```
Comment (danxmoran) [inline orchestration/templates/create-json-object.yaml:21]: This `volumeMounts` isn't needed any more, and will cause an error now that the `volumes` block has been removed.

File: orchestration/templates/create-json-object.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+          - name: archive-path
+          - name: upload-path
+      {{- include "argo.retry" . | indent 6 }}
+      script:
+        image: google/cloud-sdk:slim
+        command: [bash]
+        source: |
+        json='{"release_date":"'{{ "{{inputs.parameters.release-date}}" }}'","archive_path":"'{{ "{{inputs.parameters.archive-path}}" }}'"}'
+        echo "$json" > release_history.json
+        gsutil -m rsync -r /state/release_history.json {{ "{{inputs.parameters.upload-path}}" }}
```
Comment (danxmoran) [inline orchestration/templates/create-json-object.yaml:20]: I think the `rsync` will complain if the source path points to a file. You can use `gsutil cp` instead.

File: orchestration/templates/create-json-object.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+          - name: archive-path
+          - name: upload-path
+      {{- include "argo.retry" . | indent 6 }}
+      script:
+        image: google/cloud-sdk:slim
+        command: [bash]
+        source: |
+            json='{"release_date":"'{{ "{{inputs.parameters.release-date}}" }}'","archive_path":"'{{ "{{inputs.parameters.archive-path}}" }}'"}'
+            echo "$json" > release_history.json
+            gsutil -m rsync -r /state/release_history.json {{ "{{inputs.parameters.upload-path}}" }}
```
Comment (danxmoran) [inline orchestration/templates/create-json-object.yaml:20]: Previous comment got lost: I think `rsync` will complain if the source path isn't a directory. The input path is also mismatched from where the JSON is being written. This should work instead:
```suggestion
            gsutil cp release_history.json {{ "{{inputs.parameters.upload-path}}" }}
```

File: orchestration/templates/ingest-clinvar-archive.yaml [hunk 4]
```diff
@@ -120,7 +120,21 @@ spec:
...
+          # 6) Stage a release_history object in GCS
+          - name: stage-release-history
+            dependencies: [extract-release-date]
+            templateRef:
+              name: create-json-object
+              template: main
+            arguments:
+              parameters:
+                - name: release-date
+                  value: '{{ "/{{tasks.extract-release-date.outputs.result}}.xml" }}'
```
Comment (danxmoran) [inline orchestration/templates/ingest-clinvar-archive.yaml:132]: This should just be the release date, without the file-system stuff:
```suggestion
                  value: '{{ "{{tasks.extract-release-date.outputs.result}}" }}'
```

**Comment by quazi-broad** [comment]
Confirmed that this code is working as expected with the most recent run in the dev cluster.
`wm8cd-78d:GIT qhoque$ gsutil cat gs://broad-dsp-monster-clingen-dev-staging-storage/20200408T134747/processed/release_history/release_history.json
{"release_date":"2020-04-07","archive_path":"gs://broad-dsp-monster-clingen-dev-staging-storage/20200408T134747/raw/ClinVarRelease.xml"}`

**Pull request merged**